app/services/ect_at_school_periods/switch_training.rb

```diff
@@ -0,0 +1,85 @@
+module ECTAtSchoolPeriods
+  class IncorrectTrainingProgrammeError < StandardError; end
+  class NoTrainingPeriodError < StandardError; end
+
+  class SwitchTraining
+    include TrainingPeriodSources
+
+    def self.to_school_led(...) = new(...).to_school_led
+    def self.to_provider_led(...) = new(...).to_provider_led
+
+    def initialize(ect_at_school_period, author:, lead_provider: nil)
+      raise ArgumentError, "an `ECTAtSchoolPeriod` must be provided" unless
+        ect_at_school_period.is_a?(ECTAtSchoolPeriod)
+
+      @ect_at_school_period = ect_at_school_period
+      @training_period = ect_at_school_period.current_or_next_training_period
+
+      raise NoTrainingPeriodError unless @training_period
+
+      @school = @ect_at_school_period.school
+      @lead_provider = lead_provider.presence || @training_period.lead_provider
+      @author = author
+    end
+
+    def to_school_led
+      raise IncorrectTrainingProgrammeError if
+        @ect_at_school_period.school_led_training_programme?
+
+      ActiveRecord::Base.transaction do
+        finish_or_destroy_existing_training_period!
+        create_school_led_training_period!
+      end
+    end
+
+    def to_provider_led
+      raise IncorrectTrainingProgrammeError if
+        @ect_at_school_period.provider_led_training_programme?
+
+      ActiveRecord::Base.transaction do
+        finish_training_period!
+        create_provider_led_training_period!
+      end
+    end
+
+  private
+
+    attr_reader :school, :lead_provider
+
+    def finish_or_destroy_existing_training_period!
+      if @training_period.school_partnership.present?
+        finish_training_period!
+      else
+        @training_period.destroy!
+      end
+    end
+
+    def finish_training_period!
+      TrainingPeriods::Finish.ect_training(
+        training_period: @training_period,
+        ect_at_school_period: @ect_at_school_period,
+        finished_on:,
+        author: @author
+      ).finish!
```

Comment:
This is blowing up for me when I change from school-led to provider-led

<img width="1500" height="411" alt="image" src="https://github.com/user-attachments/assets/8934828c-e448-45db-83bc-7caf0802fa65" />

Reply:
I fixed a similar bug for mentors in #1364 , where we allowed same day handovers. I think it makes sense to align ECTs with that behaviour, but worth running it past @mason-emily just to be sure.

Reply:
We agreed to allow same day handovers. This has been fixed in https://github.com/DFE-Digital/register-early-career-teachers-public/pull/1382/commits/7a0c227907655b0323a3d889ad482a5bc9a99ec8 👌🏼